setup.py

```diff
@@ -348,7 +349,8 @@ def parse_setuppy_commands():
     # If we got here, we didn't detect what setup.py command was given
     import warnings
     warnings.warn("Unrecognized setuptools command, proceeding with "
-                  "generating Cython sources and expanding templates",
+                  "generating Cython sources and expanding templates.\n"
+                  "arguments: {}".format(args),
```

Comment:
Should we turn this into an error?

Reply:
An error does seem more appropriate, but maybe it breaks something down stream. There is probably a reason it wasn't used originally.

Reply:
Do downstream libraries call our setup.py? I would think that is numpy-specific

Reply:
Downstream folks can run setup commands, for instance `python setup.py version` will show the current NumPy version string. OTOH, no one seems to notice the warning, we didn't :) Let's make it an error and fix it if it causes problems.

Reply:
Error sounds fine to me, current version in the PR looks good.